packages/editor/src/client/atoms/permissionControl.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {ComponentType, PropsWithChildren} from 'react'
+import {AuthContext} from '../authContext'
+import {useTranslation} from 'react-i18next'
+
+interface PermissionControlProps {
+  requiredPermission: string
+  showMessage?: boolean
+}
+
+export function PermissionControl({
+  children,
+  requiredPermission,
+  showMessage
+}: PropsWithChildren<PermissionControlProps>) {
+  const {t} = useTranslation()
+  return (
+    <AuthContext.Consumer>
+      {value => (
+        <>
+          {value.session?.roles?.some(role =>
+            role.permissions.some(permission => permission.id === requiredPermission)
+          ) ? (
+            children
+          ) : showMessage ? (
+            <p>{t('permissions.noAccess')}</p>
+          ) : null}
+        </>
+      )}
+    </AuthContext.Consumer>
+  )
+}
+
+export const createCheckedPermissionComponent = (permission: string, showMessage?: boolean) => <
```

Comment:
Do we want to have showMessage default to true? Would make sense I think.

Reply:
Definitely a better idea.